src/db/MongoDB.js

```diff
@@ -64,94 +68,171 @@ class MongoDB {
+    }
+
+    async changeTabSize(documentLink, newTabSize) {
+        if (Number.isInteger(newTabSize)) {
+            return this.changeParam(documentLink, 'tab', newTabSize);
+        }
+    }
+
+    async changeLanguage(documentLink, newLanguage) {
+        if (["python"].includes(newLanguage)) {
+            return this.changeParam(documentLink, 'language', newLanguage);
+        }
+    }
+
+    async addNewEditors(documentLink, newEditorsId) {
+        try {
+            await this.documentsCollection.updateOne({documentLink: documentLink}, {$addToSet: {editors: newEditorsId}});
+            return 'Success';
+        } catch (err) {
+            if (configs.DEBUG) {
+                console.error(error);
+            }
+        }
+    }
+
+    async updateLastViewedDate(documentLink) {
+        return this.changeParam(documentLink, 'lastViewedDate', Date.now());
+    }
+
+    async deleteOldDocuments(days) {
+        const oldTimestamp = Date.now() - 1000 * 60 * 60 * 24 * days;
+        return this.documentsCollection.deleteMany({'lastViewedDate': {$lt : oldTimestamp} });
+    }
+
     async applyRequests (documentLink, requests) {
         // TODO look to use bulk write
+        let success = true;
         try {
+            // Avoid too many requests
+            requests = requests.slice(0, 50);
             for (let request of requests) {
                 let requestType = request.type;
                 let data = request.data;
+                let results = ""
                 switch (requestType) {
                     case 'set-line':
-                        await this.setLine(documentLink, data.id, data.content);
+                        results = await this.setLine(documentLink, data.id, data.content);
+                        if (!results) success = false;
                         break;
                     case 'new-line':
-                        await this.newLine(documentLink, data.previous, data.id, data.content);
+                        results = await this.newLine(documentLink, data.previous, data.id, data.content);
+                        if (!results) success = false;
                         break;
                     case 'delete-line':
-                        await this.deleteLine(documentLink, data.id);
+                        results = await this.deleteLine(documentLink, data.id);
+                        if (!results) success = false;
                         break;
                 }
             }
+            return success;
         } catch (err) {
             if (configs.DEBUG) {
                 console.error('Error when applying requests');
             }
-            throw new Error(err);
         }
     }
 }
 
 function getDB () {
-    db = new MongoDB(
-        configs.DB_CONFIG.DB_USERNAME,
-        configs.DB_CONFIG.DB_PASSWORD,
-        configs.DB_CONFIG.DB_HOST,
-        configs.DB_CONFIG.DB_DATABASE,
-        configs.DB_CONFIG.DB_PORT
-    );
+    db = new MongoDB(configs.DB_URL);
```

Comment:
⚠️ This is (also) a potentially breaking change ⚠️ 
--------------------------------------------------
How do we change the config in production?

Reply:
New description resolves this